test_manager/test_wrapper/test_wrapper.go

```diff
@@ -0,0 +1,156 @@
+package main
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"net/url"
+	"os"
+	"os/exec"
+
+	"cloud.google.com/go/storage"
+	junitFormatter "github.com/jstemmer/go-junit-report/formatter"
+	junitParser "github.com/jstemmer/go-junit-report/parser"
+)
+
+const (
+	metadataURLPrefix   = "http://metadata.google.internal/computeMetadata/v1/instance/attributes/"
+	testResultObject    = "outs/junit_go-test.xml"
+	testBinaryLocalPath = "image_test"
+	artifactPath        = "/artifact"
+	workDir             = "/workspace"
+)
+
+func main() {
+	ctx := context.Background()
+	client, err := storage.NewClient(ctx)
+	if err != nil {
+		log.Fatalf("failed to create cloud storage client: %v", err)
+	}
+
+	testBinaryURL, err := getMetadataAttribute("_test_binary_url")
+	if err != nil {
+		log.Fatalf("failed to get metadata _test_binary_url: %v", err)
+	}
+
+	testRun, err := getMetadataAttribute("_test_run")
+
+	var testArguments = []string{"-test.v"}
+	if testRun != "" {
+		testArguments = append(testArguments, "-test.run", testRun)
+	}
+
+	if err = os.Mkdir(workDir+artifactPath, 0755); err != nil {
+		log.Fatalf("failed to create artifact dir: %v", err)
+	}
+	if err = os.Mkdir(workDir, 0755); err != nil {
+		log.Fatalf("failed to create work dir: %v", err)
+	}
+	if err = os.Chdir(workDir); err != nil {
```

Comment:
nitpick but if main thread changes dir mid-work, it is a bit fragile. usually we will download-TO, and then execute-IN, where functions do the chdir if needed. probably not even needed then,though, exec should take a directory argument. these models are just best practice, make it easier to add e.g. goroutine support later, etc. and in case we ever want to write more files, we know we are not in same dir as the child process, nobody can step on any toes

Reply:
fyi the `exec.Cmd` struct has a `Dir` attribute that you can set for working dir

Reply:
Add a Dir parameter in executeCommand() to set workDir

Reply:
unresolving, what we want is a dir param in executeCmd and a dst param in downloadGCSObject, then remove chdir from main()